samza-core/src/main/java/org/apache/samza/startpoint/StartpointManager.java

```diff
@@ -158,8 +158,25 @@ public void writeStartpoint(SystemStreamPartition ssp, TaskName taskName, Startp
    * @return {@link Optional} of {@link Startpoint} for the {@link SystemStreamPartition}.
    *         It is empty if it does not exist or if it is too stale.
    */
+  @VisibleForTesting
   public Optional<Startpoint> readStartpoint(SystemStreamPartition ssp) {
-    return readStartpoint(ssp, null);
```

Comment:
Revert the change on line line 162 calling `readStartpoint(ssp, null);` will still do the intended, do not need line 163-165

Reply:
Is used in numerous tests, so need it.

Reply:
You are not changing the function signature, so the test should still work when you invoke `readStartpoint(ssp, null)` from here what am I missing?

Reply:
The fact that null as task-name maps to "only-ssp" is an internal of the class, i.e., the unit. 
Unit-tests test the unit not internal nitty gritties of the unit.